surefire-providers/surefire-junit3/src/test/java/org/apache/maven/surefire/junit/JUnitTestSetTest.java

```diff
@@ -54,6 +58,32 @@ public void testExecuteSuiteClass()
                       succeededTests.get( 0 ).getName() );
     }
 
+    public void testSystemManager()
+    {
+        float javaVersion = Float.parseFloat( JavaVersion.JAVA_RECENT.toString() );
+        boolean isDeprecated = javaVersion >= 17;
+        try
+        {
+            JUnit3Provider.setSystemManager( "java.lang.SecurityManager" );
+
+            if ( isDeprecated )
+            {
+                fail();
+            }
+
+            Object sm = invokeGetter( System.class, null, "getSecurityManager" );
```

Comment:
System.getSecurityManager() is public ... why  by reflections

Reply:
@slawekjaranowski
SecurityManager can be deleted in any major version.